packages/blockchain-wallet-v4-frontend/src/data/prices/slice.ts

```diff
@@ -0,0 +1,77 @@
+/* eslint-disable no-param-reassign */
+import { createSlice, PayloadAction } from '@reduxjs/toolkit'
+import { keys, map, mergeAll } from 'ramda'
+
+import Remote from 'blockchain-wallet-v4/src/remote/remote'
+
+import { CoinPricesRequestType, PricesStateType } from './types'
+
+const createPricesKvPairs = (prices) => {
+  return mergeAll(
+    map(
+      (x) => ({
+        // @ts-ignore
+        [x.split('-')[0]]: prices[x].price
+      }),
+      keys(prices)
+    )
+  )
+}
+
+const initialState: PricesStateType = {
+  current: Remote.NotAsked,
+  previousDay: Remote.NotAsked
+}
+
+// This combines the action with the reducer in one slice
+const pricesSlice = createSlice({
+  initialState,
+  name: 'prices',
+  reducers: {
+    // This optionally loads a action payload because you cant make your own action creators
+    // This was done because the request payload was optional in the original prices/actions.ts
+    fetchCoinPrices: {
+      prepare: (request?: CoinPricesRequestType) => ({ payload: request || {} }),
+      reducer: (state, action: PayloadAction<CoinPricesRequestType>) => {}
+    },
```

Comment:
can just be `fetchCoinPrices: (state, action: PayloadAction<CoinPricesRequestType>) => {},`

Reply:
can remove the comment lines as well